e2e/test/iothub/messaging/MessageReceiveE2ETests.cs

```diff
@@ -627,6 +650,34 @@ private async Task ReceiveSingleMessageAsync(TestDeviceType type, Client.Transpo
             await serviceClient.CloseAsync().ConfigureAwait(false);
         }
 
+        private async Task Mqtt_ReceiveSingleMessageWithCancelledAsync(TestDeviceType type)
+        {
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(Logger, s_devicePrefix, type).ConfigureAwait(false);
```

Comment:
I think this test can be simplified to just have a new device client instance make a deviceClient.ReceiveAsync(cts.Token) call and assert on the exception, right? Meaning, we don't even need the service client side of the flow, right?

Reply:
yes makes sense. Let me simply it.